src/org/labkey/test/components/ui/grids/GridBar.java

```diff
@@ -416,15 +416,24 @@ else if(onSamplePage)
 
     public GridBar searchFor(String searchStr)
     {
-        elementCache().searchBox.set(searchStr);
-        elementCache().searchBox.getComponentElement().sendKeys(Keys.ENTER);
+        _queryGrid.doAndWaitForUpdate(()->
+        {
+            elementCache().searchBox.set(searchStr);
+            elementCache().searchBox.getComponentElement().sendKeys(Keys.ENTER);
+        });
```

Comment:
I think there was a similar change to `searchFor` in the last update to this component and it had to get rolled back. The grid wont update if the search box doesn't actually change (I think this needs something similar to the `if` in `clearSearch`)

Reply:
As discussed, added call to clearSearch before setting the new value.